src/sage/homology/chain_complex_morphism.py

```diff
@@ -524,58 +525,58 @@ def __eq__(self,x):
+            sage: S1 = simplicial_complexes.Sphere(1)                                   # optional - sage.graphs
+            sage: H = Hom(S1, S1)                                                       # optional - sage.graphs
+            sage: flip = H({0:0, 1:2, 2:1})                                             # optional - sage.graphs
+            sage: flip.associated_chain_complex_morphism().is_surjective()              # optional - sage.graphs
             True
 
-            sage: pt = simplicial_complexes.Simplex(0)
-            sage: inclusion = Hom(pt, S1)({0:2})
-            sage: inclusion.associated_chain_complex_morphism().is_surjective()
+            sage: pt = simplicial_complexes.Simplex(0)                                  # optional - sage.graphs
+            sage: inclusion = Hom(pt, S1)({0:2})                                        # optional - sage.graphs
+            sage: inclusion.associated_chain_complex_morphism().is_surjective()         # optional - sage.graphs
             False
-            sage: inclusion.associated_chain_complex_morphism(cochain=True).is_surjective()
+            sage: inclusion.associated_chain_complex_morphism(cochain=True).is_surjective()     # optional - sage.graphs
             True
         """
         m = self.to_matrix()
         return m.rank() == m.nrows()
 
     def is_injective(self):
         """
-        True if this map is injective.
+        Return ``True`` if this map is injective.
 
         EXAMPLES::
 
-            sage: S1 = simplicial_complexes.Sphere(1)
-            sage: H = Hom(S1, S1)
-            sage: flip = H({0:0, 1:2, 2:1})
-            sage: flip.associated_chain_complex_morphism().is_injective()
+            sage: S1 = simplicial_complexes.Sphere(1)                                   # optional - sage.graphs
+            sage: H = Hom(S1, S1)                                                       # optional - sage.graphs
+            sage: flip = H({0:0, 1:2, 2:1})                                             # optional - sage.graphs
+            sage: flip.associated_chain_complex_morphism().is_injective()               # optional - sage.graphs
             True
 
-            sage: pt = simplicial_complexes.Simplex(0)
-            sage: inclusion = Hom(pt, S1)({0:2})
-            sage: inclusion.associated_chain_complex_morphism().is_injective()
+            sage: pt = simplicial_complexes.Simplex(0)                                  # optional - sage.graphs
+            sage: inclusion = Hom(pt, S1)({0:2})                                        # optional - sage.graphs
+            sage: inclusion.associated_chain_complex_morphism().is_injective()          # optional - sage.graphs
             True
-            sage: inclusion.associated_chain_complex_morphism(cochain=True).is_injective()
+            sage: inclusion.associated_chain_complex_morphism(cochain=True).is_injective()      # optional - sage.graphs
             False
```

Comment:
You have already done it, but it seems that you may have tagged the file with `sage.graphs`. Is there a reason that you did not? Is there a criterion when you decide on that?

Reply:
I have to admit that I don't have a clear general criterion for this decision.

But in this file, there are substantial tests that construct the `ChainComplex` directly without going through a simplicial complex (which the `sage.graphs` is denoting here by association). And (just speculating) it is conceivable that there would be added value in adding more examples of this type to the file.

Reply:
OK.